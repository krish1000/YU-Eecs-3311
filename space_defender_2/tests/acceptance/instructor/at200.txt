--Test friendly projectile moves into friendly projectile.
toggle_debug_mode
play(5,30,1,1,1,1,1)

setup_select(4)
setup_next(5)

move(C,2)
fire
move(C,10)
fire
pass
pass
