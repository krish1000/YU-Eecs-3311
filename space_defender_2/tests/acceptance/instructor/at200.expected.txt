  state:not started, normal, ok
  Welcome to Space Defender Version 2.
->toggle_debug_mode
  state:not started, debug, ok
  In debug mode.
->play(5,30,1,1,1,1,1)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_select(4)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Rocket
->setup_next(5)
  state:in game(0.0), debug, ok
  Starfighter:
    [0,S]->health:70/70, energy:60/60, Regen:11/2, Armour:3, Vision:14, Move:9, Move Cost:5, location:[C,1]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(C,2)
  state:in game(1.0), debug, ok
  Starfighter:
    [0,S]->health:70/70, energy:55/60, Regen:11/2, Armour:3, Vision:14, Move:9, Move Cost:5, location:[C,2]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [C,1] -> [C,2]
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(2.0), debug, ok
  Starfighter:
    [0,S]->health:60/70, energy:57/60, Regen:11/2, Armour:3, Vision:14, Move:9, Move Cost:5, location:[C,2]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
    [-1,*]->damage:100, move:1, location:[B,1]
    [-2,*]->damage:100, move:1, location:[D,1]
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [C,2].
      A friendly projectile(id:-1) spawns at location [B,1].
      A friendly projectile(id:-2) spawns at location [D,1].
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(C,10)
  state:in game(3.0), debug, ok
  Starfighter:
    [0,S]->health:70/70, energy:19/60, Regen:11/2, Armour:3, Vision:14, Move:9, Move Cost:5, location:[C,10]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
    [-1,*]->damage:100, move:2, location:[B,2]
    [-2,*]->damage:100, move:2, location:[D,2]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [B,1] -> [B,2]
    A friendly projectile(id:-2) moves: [D,1] -> [D,2]
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [C,2] -> [C,10]
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(4.0), debug, ok
  Starfighter:
    [0,S]->health:60/70, energy:21/60, Regen:11/2, Armour:3, Vision:14, Move:9, Move Cost:5, location:[C,10]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
    [-1,*]->damage:100, move:4, location:[B,4]
    [-2,*]->damage:100, move:4, location:[D,4]
    [-3,*]->damage:100, move:1, location:[B,9]
    [-4,*]->damage:100, move:1, location:[D,9]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [B,2] -> [B,4]
    A friendly projectile(id:-2) moves: [D,2] -> [D,4]
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [C,10].
      A friendly projectile(id:-3) spawns at location [B,9].
      A friendly projectile(id:-4) spawns at location [D,9].
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  *  _  _  _  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  *  _  _  _  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->pass
  state:in game(5.0), debug, ok
  Starfighter:
    [0,S]->health:70/70, energy:25/60, Regen:11/2, Armour:3, Vision:14, Move:9, Move Cost:5, location:[C,10]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
    [-1,*]->damage:100, move:8, location:[B,8]
    [-2,*]->damage:100, move:8, location:[D,8]
    [-3,*]->damage:100, move:2, location:[B,10]
    [-4,*]->damage:100, move:2, location:[D,10]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [B,4] -> [B,8]
    A friendly projectile(id:-2) moves: [D,4] -> [D,8]
    A friendly projectile(id:-3) moves: [B,9] -> [B,10]
    A friendly projectile(id:-4) moves: [D,9] -> [D,10]
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) passes at location [C,10], doubling regen rate.
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  *  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  *  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->pass
  state:in game(6.0), debug, ok
  Starfighter:
    [0,S]->health:70/70, energy:29/60, Regen:11/2, Armour:3, Vision:14, Move:9, Move Cost:5, location:[C,10]
      Projectile Pattern:Rocket, Projectile Damage:100, Projectile Cost:10 (health)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
    [-1,*]->damage:200, move:16, location:[B,16]
    [-2,*]->damage:200, move:16, location:[D,16]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [B,8] -> [B,16]
      The projectile collides with friendly projectile(id:-3) at location [B,10], combining damage.
    A friendly projectile(id:-2) moves: [D,8] -> [D,16]
      The projectile collides with friendly projectile(id:-4) at location [D,10], combining damage.
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) passes at location [C,10], doubling regen rate.
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
